pennylane/data/data_manager/progress/_default/__init__.py

```diff
@@ -0,0 +1,185 @@
+        self._curr_longest_description = 0
+
+    def __enter__(self) -> "DefaultProgress":
+        self._active = True
+
+        return self
+
+    def __exit__(
+        self,
+        exc_type: Optional[type[BaseException]],
+        exc_value: Optional[BaseException],
+        traceback: Any,
+    ) -> None:
+        self._print_final()
+        self._active = False
+
+    def add_task(self, description: str, total: Optional[float] = None) -> int:
+        """Add a task."""
+        description = _truncate(description, self._term_info.description_len_max)
+        self.tasks.append(Task(description=description, total=total))
+
+        self._curr_longest_description = max(self._curr_longest_description, len(description))
+        self.refresh()
+
+        return len(self.tasks) - 1
+
+    def refresh(self, task_id: Optional[int] = None):
+        """Refresh display liens for one or all tasks."""
+        if task_id is None:
+            self._task_display_lines.clear()
+            self._task_display_lines.extend(
+                self._get_task_display_line(task)
+                for task in self.tasks[: self._term_info.max_display_lines]
+            )
+
+            if len(self.tasks) > self._term_info.max_display_lines:
+                self._task_display_lines.append(f"{term.erase_line()}...")
+
+        elif task_id < self._term_info.max_display_lines:
+            self._task_display_lines[task_id] = self._get_task_display_line(self.tasks[task_id])
+
+    def update(
+        self,
+        task_id: int,
+        completed: Optional[float] = None,
+        total: Optional[float] = None,
+        advance: Optional[float] = None,
+    ):
+        """Update task with given ``task_id`` and refresh its progress.
+
+        Args:
+            task_id: ID of task
+            completed: Set the completed state of the task
+            total: Set the total for the task
+            advance: Advance the completion state of the task
+        """
+        task = self.tasks[task_id]
+        task.update(advance=advance, completed=completed, total=total)
+
+        self.refresh(task_id)
+
+        if self._active:
+            self._print()
+
+    def _get_task_display_line(self, task: Task) -> str:
+        """Get display line for the task."""
+        if task.total is None:
+            progress_column = f"{task.completed / 1e6:.2f} MB"
+        else:
+            progress_column = f"{task.completed / 1e6:.2f}/{task.total / 1e6:.2f} MB"
+
+        display = _truncate(
+            f"{task.description.ljust(self._curr_longest_description + 1)}" f"{progress_column}",
+            self._term_info.columns,
+        )
+
+        return f"{term.erase_line()}{display}"
+
+    def _print_final(self):
+        """Print all tasks without any terminal control. Should be called when
+        progress is complete."""
+        print(*(self._get_task_display_line(task) for task in self.tasks), sep="\n", flush=True)
+
+    def _print(self):
```

Comment:
this doesn't seem to update the previous lines properly. in the default Mac terminal, I ran the following and it pushed past the top:
```pycon
>>> import pennylane as qml
>>> c2h4 = qml.data.load("qchem", molname="C2H4")
```

Reply:
Had a similar issue on Ubuntu:
![LoadProgressBehavior](https://github.com/user-attachments/assets/a705d8c9-82a5-430b-971a-436230cb2b71)

Reply:
@DSGuala, @timmysilv should work now - the terminal control was bugged when there was only one task